### PR TITLE
Update bookmark relations API

### PR DESCRIPTION
**Pull request by ityankel**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/adx-documentation-pr/wiki/Overall-basic-flow) for the workflow.
- [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [ ] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

**Comment by ityankel** [comment]
This pull request includes a breaking change in the request and response models for the microsoft.securityinsights/bookmarks/relations nested resource type. 
We currently have 3 resources types that have a nested resource type of 'relations' - incidents, entities and bookmarks - but bookmarks has a separate structure than the other 2. This pull request will align the 3 APIs.

Bookmark relations API currently only gets traffic from our test environments so this change isn't expected to affect any customers.

File: specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2019-01-01-preview/SecurityInsights.json [hunk 1]
```diff
@@ -1527,20 +1527,20 @@
...
             "description": "OK",
             "schema": {
-              "$ref": "#/definitions/BookmarkRelation"
+              "$ref": "#/definitions/Relation"
             }
           },
           "201": {
             "description": "Created",
             "schema": {
-              "$ref": "#/definitions/BookmarkRelation"
```
Comment (ryansbenson) [inline specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2019-01-01-preview/SecurityInsights.json]: BookmarkRelation and BookmarkRelationList aren't needed in the definition list anymore.

File: specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2019-01-01-preview/SecurityInsights.json [hunk 2]
```diff
@@ -1437,7 +1437,7 @@
             }
           }
         },
-        "x-ms-odata": "#/definitions/BookmarkRelation",
+        "x-ms-odata": "#/definitions/Relation",
```
Comment (ryansbenson) [inline specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2019-01-01-preview/SecurityInsights.json]: Line: 6921 - In the definition of 'relation', 'relatedResourceName' should be readonly as 'relatedResourceId' is the only one required.

**Review by fengzhou-msft — APPROVED** [review]

**Review by dosegal — APPROVED** [review]

**Comment by ityankel** [comment]
@fengzhou-msft , this PR is approved and ARM signed off. Can you please merge it?

**Comment by ityankel** [comment]
@akning-ms , the breaking changes in this PR were approved by ARM more than a month ago and I'm waiting for it to be merged since then. In the last PR where I had breaking changes you were the one that could force merge them. Can you please do the same here?
Thanks!

**Comment by akning-ms** [comment]
Hi @ityankel, Can you help fix Model validation failure first?

**Comment by ityankel** [comment]
@akning-ms , I fixed the model validation. Please merge when you can

**Pull request merged**